### ISSUE TITLE
Example of how to use parameters

### ISSUE THREAD

**Issue by othbert** [stefandoorn/sitemap-plugin#39]
Hi, thanks for the plugin, all looks great except can't quite work out how to use the parameters on static routes effectively.

We have, for example, a single route for static pages that takes a parameter "page" and accepts several strings in order to generate some static content pages... imagine "/terms , /about , /info", those kinds of things. Could you please give some pointers on how would it be possible to specify in the configuration that these pages should be added to the sitemap? 
The closest I've got to it working with the configuration is something like
```
parameters:
    - ['page', ['terms', 'about', 'info']]
```

But this gives:
```
InvalidConfigurationException
Unrecognized options "0, 1" under "sylius_sitemap.static_routes.0.parameters.0"
```

Thanks!

**Comment by stefandoorn** [comment]
Hi @othbert. I think this is similar to #34. Recently a PR was submitted (#38) to resolve this issue, but I haven't been able to test it yet. Could you test that PR to see if it resolves your issue?

**Comment by othbert** [comment]
Hi there! Yup, sorry, I was going to submit as a comment on that, but thought twice about it.

I did test that PR and I think it's pointed me in a better direction... However, I'd love to see an example of the intended setup of parameters for a route, as I think I may be mistaking how this should work.

Anyway, I'm now supplying the same route alias several times, changing the page parameter each time, this seems to work better along with the fix in the PR, however I end up with duplicated entries in the resultant sitemap.

configuration:
```
    static_routes:
        -
            route: static_content_page
            parameters:
                page: 'terms'
            locales: ['en','de','es']
        -
            route: static_content_page
            parameters:
                page: 'about'
            locales: ['en','de','es']
        -
            route: static_content_page
            parameters:
                page: 'info'
            locales: ['en','de','es']
```

Gives me the entries for these routes, but after it's listed them all, it gives me the output again so it ends up with .../en/terms etc. following .../es/info

**Comment by stefandoorn** [comment]
Could you paste the resulting XML? It could be that it's according to Google standards, even though it feels duplicate.

**Comment by othbert** [comment]
```

<?xml version="1.0" encoding="UTF-8"?>
<urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9" xmlns:xhtml="http://www.w3.org/1999/xhtml">
	<url>
		<loc>http://test.local/de/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/terms</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/terms"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/terms"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/terms"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/about</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/about"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/about"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/about"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/en/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/de/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
	<url>
		<loc>http://test.local/es/info</loc>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="en" href="http://test.local/en/info"/>
		<xhtml:link rel="alternate" hreflang="de" href="http://test.local/de/info"/>
		<xhtml:link rel="alternate" hreflang="es" href="http://test.local/es/info"/>

		<changefreq>weekly</changefreq>

		<priority>0.3</priority>

	</url>
</urlset>
```

I just realised there are some duplications with the German language too, not entirely sure why that is occurring, possible it is being determined as the default locale.

**Comment by stefandoorn** [comment]
Indeed a bit too much duplication.. not directly sure what is the issue here. There is a Static Url Provider handling these routes and URL generation. I'm not able to look into this in the upcoming days, so if you could poke a bit around in there and maybe figure out what's the issue, that would be highly appreciated!

**Comment by stefandoorn** [comment]
Oeps, this one shouldn't be closed. Due to the other PR being merged, it's now probably easier to test and identify this issue.

**Comment by stefandoorn** [comment]
PR in #40 should fix this issue. Would you be able to confirm @othbert?

**Comment by othbert** [comment]
Sorry, had a busy weekend and didn't get a chance to look at this. Thanks for the improvement, that helped indeed with one part, alternate urls for the same locale as the current <loc> object are now not being reported. So no more duplicates where domain/de/path has an alternate also of domain/de/path.

However, I'm still seeing the entire list of static routes being duplicated. If I get a chance this week I'll have a look into it too.
Thanks!

**Comment by stefandoorn** [comment]
Reopening, part of it it not fixed yet.

**Comment by stefandoorn** [comment]
@othbert Weird.. Just to check, these 3 locales are also activated for the current channel you are requesting the sitemap in?

**Comment by othbert** [comment]
That's interesting, yeah, we have a setup with several channels (more than 2) but all the locales are available in them. However, we are doing some non-standard stuff with the channel resolving, I'll take a closer look at it again tomorrow.

**Comment by stefandoorn** [comment]
Curious to see if you find/found something, @othbert.

**Comment by othbert** [comment]
Hi Stefan, sorry I've been working on other stuff recently and haven't had a chance to get back into this at all yet. Hopefully this weekend!

**Comment by stefandoorn** [comment]
@othbert any updates?

**Comment by othbert** [comment]
Sorry to get back so late, unfortunately have been looking elsewhere. Will be revisiting the sitemap handling sometime in the next few months as part of SEO updates I guess as there are plans to add dynamically generated pages. Afraid I won't really be looking at it much until then though. If it's all working in your test setup, again, I fear it may just be down to our particular channel resolving strategy and therefore not even related.

**Comment by stefandoorn** [comment]
@othbert Thanks for the update. I'll close it for now, please reopen it if you find any issue if you have time again.

**Comment by othbert** [comment]
Well, a whole year later, the task fell upon me again to look into what was causing the duplication. Happy to report it was just a dodgy misconfiguration on our part, but not the channel resolving in the end. Details below for posterity and anyone that may end up here with similar issues...

---
Naturally, with the yaml structured as it is, new routes are appended to the static_routes if the configuration file happens to be called twice, rather than overwriting the first declaration of the static_routes.

We called the sitemap.yaml from both the default configuration services and also from environment specific config in the old versions of sylius/symfony. In the new version, the issue was migrated over by having it called in both the app/config/services and from our bundle configuration. 

After tidying the configuration call, I have moved it simply into app/config/packages and removed calls to it elsewhere. This has fixed the duplication of routes in the output. Sorry it took so long to finally get back on this!
